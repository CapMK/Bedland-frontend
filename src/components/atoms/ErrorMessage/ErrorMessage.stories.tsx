import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import ErrorMessage from './ErrorMessage';

const meta = {
  title: 'Atoms/ErrorMessage',
  component: ErrorMessage,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        <BrowserRouter>{Story()}</BrowserRouter>
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof ErrorMessage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'error',
  },
};
