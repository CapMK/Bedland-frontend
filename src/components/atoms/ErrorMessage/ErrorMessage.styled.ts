import styled from 'styled-components';

export const ErrorMsg = styled.span`
  align-items: center;
  display: flex;
  height: 2.875rem;
  justify-content: start;
  width: 100%;
`;

export const ErrorRow = styled.span`
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 5px;
`;
