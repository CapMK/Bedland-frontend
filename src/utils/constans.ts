export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
