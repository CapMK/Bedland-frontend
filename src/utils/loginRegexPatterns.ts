import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
