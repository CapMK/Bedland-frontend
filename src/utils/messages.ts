const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
