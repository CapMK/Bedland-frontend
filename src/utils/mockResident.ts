const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
