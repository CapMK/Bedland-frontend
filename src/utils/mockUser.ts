import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
