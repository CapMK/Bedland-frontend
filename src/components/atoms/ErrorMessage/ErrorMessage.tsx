import { FC, PropsWithChildren } from 'react';

import Icon from '../Icon/Icon';
import Typography from '../Typography/Typography';
import * as Styled from './ErrorMessage.styled';

const ErrorMessage: FC<PropsWithChildren> = ({ children }) => (
  <Styled.ErrorRow>
    <div style={{ display: 'flex' }}>
      <Icon name='alert' color='redLight' size={20} isActive={false} />
    </div>
    <Typography variant='dataInputsAndTooltips' color='redLight'>
      {children}
    </Typography>
  </Styled.ErrorRow>
);

export default ErrorMessage;
