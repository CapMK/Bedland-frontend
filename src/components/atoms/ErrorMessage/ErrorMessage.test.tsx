import { renderWithProviders } from 'utils/test.utils';
import { cleanup } from '@testing-library/react';

import ErrorMessage from './ErrorMessage';

describe('ErrorMessage', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const snapshot = renderWithProviders(<ErrorMessage>Error message</ErrorMessage>, {});
    expect(snapshot).toMatchSnapshot();
  });
});
