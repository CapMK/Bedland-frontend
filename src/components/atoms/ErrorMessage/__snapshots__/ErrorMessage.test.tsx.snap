// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 5px;
}

<body>
    <div>
      <span
        class="c0"
      >
        <div
          style="display: flex;"
        >
          <svg
            data-testid="alert-icon"
            fill="none"
            height="20"
            viewBox="0 0 21 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-alert"
            />
            <path
              d="M10.5 17.5C14.6421 17.5 18 14.1421 18 10C18 5.85786 14.6421 2.5 10.5 2.5C6.35786 2.5 3 5.85786 3 10C3 14.1421 6.35786 17.5 10.5 17.5Z"
              stroke="#FF4438"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
            <path
              d="M10.5001 14.1666C10.9603 14.1666 11.3334 13.7935 11.3334 13.3333C11.3334 12.8731 10.9603 12.5 10.5001 12.5C10.0398 12.5 9.66675 12.8731 9.66675 13.3333C9.66675 13.7935 10.0398 14.1666 10.5001 14.1666Z"
              stroke="#FF4438"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
            <path
              d="M10.5 5.83331V9.99998"
              stroke="#FF4438"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </svg>
        </div>
        <p
          style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1.4; letter-spacing: 1px; color: rgb(255, 68, 56);"
        >
          Error message
        </p>
      </span>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 5px;
}

<div>
    <span
      class="c0"
    >
      <div
        style="display: flex;"
      >
        <svg
          data-testid="alert-icon"
          fill="none"
          height="20"
          viewBox="0 0 21 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-alert"
          />
          <path
            d="M10.5 17.5C14.6421 17.5 18 14.1421 18 10C18 5.85786 14.6421 2.5 10.5 2.5C6.35786 2.5 3 5.85786 3 10C3 14.1421 6.35786 17.5 10.5 17.5Z"
            stroke="#FF4438"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.66667"
          />
          <path
            d="M10.5001 14.1666C10.9603 14.1666 11.3334 13.7935 11.3334 13.3333C11.3334 12.8731 10.9603 12.5 10.5001 12.5C10.0398 12.5 9.66675 12.8731 9.66675 13.3333C9.66675 13.7935 10.0398 14.1666 10.5001 14.1666Z"
            stroke="#FF4438"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.66667"
          />
          <path
            d="M10.5 5.83331V9.99998"
            stroke="#FF4438"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.66667"
          />
        </svg>
      </div>
      <p
        style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1.4; letter-spacing: 1px; color: rgb(255, 68, 56);"
      >
        Error message
      </p>
    </span>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
